nservicebus/upgrades/5to6-moving-away-from-ibus.md

```diff
@@ -0,0 +1,82 @@
+
+snippet:5to6-bus-send-publish
+
+
+## Accessing the CurrentMessageContext
+
+Previously it was possible to access message parameters such `MessageId`, `ReplyToAddress` and the message headers via the `CurrentMessageContext` property on the `IBus` interface. These message properties are now available directly via the message handler context parameter.
+
+snippet: 5to6-messagecontext
+
+
+## Dependency Injection
+
+In previous versions of NServiceBus the `IBus` interface was automatically registered in the IoC container. In Version 6, the new context aware interfaces, for example, the `IEndpointInstance`, `IMessageSession` and `IMessageHandlerContext`, etc., will not be automatically registered in the IoC container.  
+
+When containers are used for dependency injection, if the `IBus` was previously being used in the injected custom classes:
+
+If the custom component was sending messages using the injected `IBus`from outside of message handlers, for example, in a MVC Controller class, then it is safe to use register the `IEndpointInstance` when self hosting. This interface can then be used to send messages. 
+
+However if the custom component is accessed from within message handlers then the `IMessageHandlerContext` parameter should be passed to the custom component instead of the `IEndpointInstance` interface to either send or publish messages. It is not safe to inject the `IEndpointInstance` for the following reasons: 
+
+If the `IEndpointInstance` interface is used inside a message handler to send or publish messages, then:
+
+- those messages will not participate in the same transaction scope as that of the message handler. This could result in messages dispatched or events published via the `IEndpointInstance` interface even if the message handler resulted in an exception and the operation was rolled back.
+
+- those messages will be part of the [batching operation](/nservicebus/messaging/batched-dispatch.md) and therefore will not have the benefits that Version 6 provides.
+
+- those messages will not contain any important message header information that is available via the `IHandlerMessageContext` interface parameter, e.g., CorrelationId.  
+
+
+## UnicastBus made internal
+
+
+### Accessing the builder
+
+When using the `IBuilder` interface outside the infrastructure of NServiceBus it was possible to use a hack by casting the `IBus` interface to `UnicastBus` and then accessing the `Builder` property like this:
+
+    var builder = ((UnicastBus)bus).Builder
+
+This is no longer supported. It is advised to, instead of using `IBuilder` directly, use dependency injection via the container of choice.
+
+
+### Setting the host information
+
+Control over `HostInformation` was previously done using `UnicastBus.HostInformation`. This is now done using the more explicit API to set the host identifier, see `busConfiguration.UniquelyIdentifyRunningInstance()`.
```

Comment:
@mikeminutillo - I didn't create this content. Just moved it over. What's the doco you want to link to?